Make deprecated task container methods an error (#10292)
gradle/gradle: subprojects/docs/src/docs/userguide/upgrading_version_5.adoc

The deprecated Compare Gradle Builds plugin has been removed.
Please use https://scans.gradle.com/[build scans] for build analysis and comparison.

==== Changes to the task container

The following deprecated methods on the task container now result in errors:

- `TaskContainer.add()`
- `TaskContainer.addAll()`
- `TaskContainer.remove()`
- `TaskContainer.removeAll()`
- `TaskContainer.retainAll()`
- `TaskContainer.clear()`
- `TaskContainer.iterator().remove()`

Additionally, the following deprecated functionality now results in an error:

- Replacing a task that has already been realized.

> - Replacing a registered (unrealized) task with a task that has an incompatible type (i.e. a type that is not the same type or a sub-type of the previously registered type)

Reviewer: ```suggestion
- Replacing a registered (unrealized) task with an incompatible type. A compatible type is the same type or sub-type of the registered type.
```

Author: Fixed

> - Attempting to replace a task that has not been previously been registered.

Reviewer: ```suggestion
- Replacing a task that was never registered.
```

Author: Fixed.
